applications/selfsupervised/main.py

```diff
@@ -0,0 +1,140 @@
+# Command-line options
+parser = argparse.ArgumentParser()
+lbann.contrib.args.add_scheduler_arguments(parser)
+parser.add_argument(
+    '--job-name', action='store', default='lbann_siamese', type=str,
+    help='job name', metavar='NAME')
+parser.add_argument(
+    '--pretrain', action='store', default='siamese', type=str,
+    help='pretraining model (default: siamese)')
+parser.add_argument(
+    '--num-patches', action='store', default=3, type=int,
+    help='number of patches and Siamese heads (default: 3)', metavar='NUM')
+parser.add_argument(
+    '--pretrain-epochs', action='store', default=20, type=int,
+    help='number of pretraining epochs (default: 20)', metavar='NUM')
+parser.add_argument(
+    '--batch-job', action='store_true',
+    help='submit script as batch job')
+parser.add_argument(
+    '--checkpoint-interval', action='store', default=0, type=int,
+    help='epoch frequency for checkpointing')
+args = parser.parse_args()
+
+# ==============================================
+# Setup experiment
+# ==============================================
+
+# Pretraining model
+if not args.pretrain or args.pretrain == 'siamese':
+    model1, reader1, opt1 = pretrain_siamese.setup(
+        num_patches=args.num_patches,
+        mini_batch_size=512,
+        num_epochs=args.pretrain_epochs,
+        learning_rate=0.005,
+        checkpoint_interval=args.checkpoint_interval,
+    )
+elif args.pretrain == 'supervised':
+    data_reader_file = os.path.join(current_dir, 'data_reader_imagenet.prototext')
+    model1, reader1, opt1 = classifier.setup(
+        data_reader_file=data_reader_file,
+        name='supervised',
+        num_labels=1000,
+        mini_batch_size=512,
+        num_epochs=args.pretrain_epochs,
+        learning_rate=0.1,
+        warmup_epochs=5,
+        learning_rate_drop_interval=30,
+        learning_rate_drop_factor=0.1,
+        checkpoint_interval=args.checkpoint_interval,
+    )
+else:
+    raise Exception(f'"{args.pretrain}" is an invalid pretraining model')
+model1.random_seed = random.getrandbits(32)
+
+# Fine-tuning model
+data_reader_file = os.path.join(current_dir, 'data_reader_cub.prototext')
+model2, reader2, opt2 = classifier.setup(
+    data_reader_file=data_reader_file,
+    name='finetune',
+    num_labels=200,
+    mini_batch_size=128,
+    num_epochs=500,
+    learning_rate=0.1,
+    warmup_epochs=50,
+    learning_rate_drop_interval=50,
+    learning_rate_drop_factor=0.25,
+)
+
+# ==============================================
+# Construct LBANN invocation
+# ==============================================
+
+# Initialize LBANN executable and command-line arguments
+lbann_exe = os.path.realpath(lbann.lbann_exe())
+lbann_exe = os.path.join(os.path.dirname(lbann_exe), 'lbann2')
+lbann_command = [lbann_exe]
+
+# Construct experiment directory
+experiment_dir = util.make_experiment_dir(args.job_name)
+
+# Export model prototext files
+# Note: lbann2 driver doesn't have a command-line argument to get
```

Comment:
Does this model require lbann2 as a front end?

Reply:
Yep. It first trains with the Siamese network on ImageNet, and then transfers the weights to a standard AlexNet and trains on CUB.